en/docs/develop-components/develop-services/develop-a-grpc-service.md

```diff
@@ -0,0 +1,210 @@
+
+In this guide, you will learn to use Choreo to create a service component that exposes a gRPC server implemented in Go, enabling efficient and scalable communication with any gRPC client application. No prior knowledge of the Go language is necessary to follow this guide.
+
+By following this guide, you will:
+
+- Create a simple gRPC server using a Service component.
+- Link the containerized gRPC service to the Choreo component using the Dockerfile. 
+    The greeter service has a single RPC method named `sayHello`. Following is the proto file the server and the client application are using:
+
+    ```proto
+    syntax = "proto3";
+    option go_package = "gtihub.com/wso2/choreo-sample-apps/go/grpc-greeter/pkg";
+
+    service greeter {
+      rpc SayHello (HelloRequest) returns (HelloReply) {}
+    }
+
+    message HelloRequest {
+      string name = 1;
+    }
+
+    message HelloReply {
+      string message = 1;
+    }
+    ```
+
+- Deploy the gRPC server component in Choreo. 
+- Create a gRPC client using a Service component. 
+- Deploy the gRPC client.
+- Invoke the gRPC server using a gRPC client and check the response through the Choreo log view.
+
+
+## Prerequisites
+
+1. To deploy a containerized component, you will need a GitHub account with a repository that contains a Dockerfile. Fork the [Choreo sample apps repository](https://github.com/wso2/choreo-sample-apps/), which contains the sample for this guide.
+2. The Choreo GitHub App requires the following permissions:
+    - Read access to issues and metadata.
+    - Read and write access to code, pull requests, and repository hooks.
+
+Let's get started!
+
+### Learn the repository file structure
+
+Let's familiarize ourselves with the key files in the sample greeter application. The below table gives a brief overview of the important files in the greeter service.
+
+!!! note 
+    The following file paths are relative to the path `<sample-repository-dir>/go/grpc`.
+
+|Filepath                |Description                                                                                   |
+|------------------------|----------------------------------------------------------------------------------------------|
+| main.go                | The greeter service code is written in the Golanguage.                                       |
+| greeter_client/main.go | The greeter client application is written in Go language                                        |
+| Dockerfile.server      | Dockerfile to build the container image of the server application.|
+| Dockerfile.client      | Dockerfile to build the container image of the client application|
+| .choreo/endpoints.yaml |  The choreo-specific configuration provides information about how Choreo exposes the service.|
```

Comment:
Need to capitalize C in Choreo-specific.

Reply:
Fixed